src/Init/Control/Basic.lean

```diff
@@ -20,8 +20,21 @@ def Functor.discard {f : Type u → Type v} {α : Type u} [Functor f] (x : f α)
 
 export Functor (discard)
 
+/--
+An `Alternative` functor is an `Applicative` functor that has a notion of “emptyness” or “failure”
+and a binary operation `<|>` with a notion of “collecting values” or “left-most success”.
+
+Important instances include
+* `Option`, where `failure := .none` and `<|>` returns the left-most `.some`.
+* Parser combinators typically provide an `Applicative` instance for error-handling and
+  backtracking.
+-/
+-- NB: List instance is in mathlib. Once upstreamed, add
+-- * `List`, where `failure` is the empty list and `<|>` concatenates.
 class Alternative (f : Type u → Type v) extends Applicative f : Type (max (u+1) v) where
+  /-- See `Alternative`. -/
```

Comment:
I don't think this is the hover I'd like to see.

```suggestion
  /--
  Produces an empty collection or recoverable failure.  The `<|>` operator collects values or recovers
  from failures.
  -/
```

Reply:
Are you sure that there shouldn't be at least a mention of `Alternative` here, as the place to look for more details?

Reply:
Yes, there should be!

Reply:
```suggestion
  /--
  Produces an empty collection or recoverable failure.  The `<|>` operator collects values or recovers
  from failures. See `Alternative` for more details.
  -/
```